src/pathstyle.rs

```diff
@@ -0,0 +1,214 @@
+//! Controls how a [`BezPath`] is converted to string form.
+
+use kurbo::{BezPath, PathEl, Point};
+
+#[derive(Debug, Copy, Clone)]
+pub enum PathStyle {
+    /// Make no effort to produce a compact path
+    Unoptimized,
+    /// Try to produce a compact path, applying roughly the same optimizations as [svgo convertPathData.js](https://github.com/svg/svgo/blob/main/plugins/convertPathData.js)
+    Compact,
+}
+
+impl PathStyle {
+    pub(crate) fn write_svg_path(&self, path: &BezPath) -> String {
+        match self {
+            PathStyle::Unoptimized => to_unoptimized_svg_path(path),
+            PathStyle::Compact => to_compact_svg_path(path),
+        }
+    }
+}
+
+fn round2(v: f64) -> f64 {
+    (v * 100.0).round() / 100.0
+}
+
+trait ToSvgCoord {
+    fn write_absolute_coord(&self) -> String;
+    fn write_relative_coord(&self, other: Self) -> String;
+}
+
+impl ToSvgCoord for f64 {
+    fn write_absolute_coord(&self) -> String {
+        format!("{}", round2(*self))
+    }
+
+    fn write_relative_coord(&self, other: Self) -> String {
+        format!("{}", round2(*self - other))
+    }
+}
+
+impl ToSvgCoord for Point {
+    fn write_absolute_coord(&self) -> String {
+        coord_string(*self)
+    }
+
+    fn write_relative_coord(&self, other: Self) -> String {
+        coord_string((*self - other).to_point())
+    }
+}
+
+fn coord_string(p: Point) -> String {
+    format!("{},{}", round2(p.x), round2(p.y))
+}
+
+fn add_command<T, const N: usize>(
+    svg: &mut String,
+    prefix: char,
+    coords: [T; N],
+    relative_to: Option<T>,
+) where
+    T: ToSvgCoord + Copy,
+{
+    assert!(prefix.is_ascii_uppercase());
+
+    let absolute = coords
+        .iter()
+        .map(|p| p.write_absolute_coord())
+        .collect::<Vec<_>>()
+        .join(" ");
+    let relative = relative_to.map(|rel_to| {
+        coords
+            .iter()
+            .map(|p| p.write_relative_coord(rel_to))
+            .collect::<Vec<_>>()
+            .join(" ")
+    });
+
+    if relative.as_ref().map(|s| s.len()).unwrap_or(usize::MAX) < absolute.len() {
+        svg.push(prefix.to_ascii_lowercase());
+        svg.push_str(&relative.unwrap());
+    } else {
+        svg.push(prefix);
+        svg.push_str(&absolute);
+    }
+}
+
+fn to_unoptimized_svg_path(path: &BezPath) -> String {
```

Comment:
I’m not sure if you’re planning future optimizations that will make merging more unpleasant but in the current state, it looks like the two svg path functions can be nicely combined.